cpp/src/parquet/column_writer.cc

```diff
@@ -1362,7 +1375,7 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
           *out_spaced_values_to_write +=
               def_levels[x] >= level_info_.repeated_ancestor_def_level ? 1 : 0;
         }
-        *null_count = *out_values_to_write - *out_spaced_values_to_write;
+        *null_count = batch_size - *out_values_to_write;
```

Comment:
Why did this calculation need to change?

Reply:
If this line does not change, the test cases `StatsTests::ParameterizedStatisticsTest` in the `arrow_statistics_test.cc` simply fail due to incorrect null_count. Here is why:
- `out_values_to_write`: number of non-null values only
- `out_spaced_values_to_write`: number of non-null values and null values
- `batch_size`: number of all values including non-null values, null values, and empty list (which is null from ancestor)